clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy

```diff
@@ -16,35 +16,56 @@
 
 package com.netflix.spinnaker.clouddriver.openstack.client
 
+import com.netflix.spinnaker.clouddriver.openstack.deploy.exception.OpenstackOperationException
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperations
 import org.openstack4j.api.OSClient
+import org.openstack4j.model.common.ActionResponse
 
 /**
  * Provides access to the Openstack API.
+ *
+ * TODO tokens will need to be regenerated if they are expired.
```

Comment:
How long do tokens usually last? This feels like something the underlying API should provide?